packages/core/data-transfer/src/strapi/remote/handlers.ts

```diff
@@ -91,20 +108,29 @@ export const createTransferHandler = (options: IHandlerOptions) => {
           }
         };
 
-        const teardown = async (): Promise<server.Payload<server.EndMessage>> => {
-          await verifyAuth(state.transfer?.kind);
-
+        const teardown = (): void => {
           delete state.controller;
           delete state.transfer;
+        };
+
+        const end = async (msg: client.EndCommand): Promise<server.Payload<server.EndMessage>> => {
+          await verifyAuth(state.transfer?.kind);
+
+          if (msg.params.transferID !== state.transfer?.id) {
+            throw new ProviderTransferError('Bad transfer ID provided');
+          }
+
+          teardown();
 
           return { ok: true };
         };
 
         const init = async (
           msg: client.InitCommand
         ): Promise<server.Payload<server.InitMessage>> => {
-          // TODO: this only checks for this instance of node: we should consider a database lock
-          if (state.controller) {
+          // TODO: For push transfer, we'll probably have to trigger a
+          // maintenance mode to prevent other transfer at the same time.
+          if (state.transfer && state.controller) {
```

Comment:
Should this be || to be safe?

Reply:
TBH we should never end up in a situation where one is defined and the other isn't, but yeah I can transform that into an or

Reply:
Done